components/RemoveProperty.coffee

```diff
@@ -1,46 +1,43 @@
 noflo = require 'noflo'
-_ = require 'underscore'
-
-class RemoveProperty extends noflo.Component
-  icon: 'ban'
-  constructor: ->
-    @properties = []
-    @inPorts = new noflo.InPorts
-      in:
-        datatype: 'object'
-        description: 'Object to remove properties from'
-      property:
-        datatype: 'string'
-        description: 'Properties to remove (one per IP)'
-      reset:
-        datatype: 'bang'
-        description: 'Clear the list of properties to remove'
-    @outPorts = new noflo.OutPorts
-      out:
-        datatype: 'object'
-        description: 'Object forwarded from input'
-
-    @inPorts.property.on 'data', (data) =>
-      @properties.push data
-    @inPorts.reset.on 'data', =>
-      @properties = []
-
-    @inPorts.in.on 'begingroup', (group) =>
-      @outPorts.out.beginGroup group
-    @inPorts.in.on 'data', (data) =>
-      return unless @properties.length
-      @outPorts.out.send @removeProperties data
-    @inPorts.in.on 'endgroup', =>
-      @outPorts.out.endGroup()
-    @inPorts.in.on 'disconnect', =>
-      @outPorts.out.disconnect()
-
-  removeProperties: (object) ->
+
+clone = (obj) ->
+  return obj if obj is null or typeof obj isnt 'object'
+  temp = new obj.constructor()
+  for key of obj
+    temp[key] = clone obj[key]
+  temp
+
+exports.getComponent = ->
+  c = new noflo.Component
+  c.icon = 'ban'
+
+  c.inPorts = new noflo.InPorts
+    in:
+      datatype: 'object'
+      description: 'Object to remove properties from'
+      required: true
+    property:
+      datatype: 'string'
+      description: 'Properties to remove (one per IP)'
+      required: true
+
+  c.outPorts = new noflo.OutPorts
+    out:
+      datatype: 'object'
+      description: 'Object forwarded from input'
+
+  c.process (input, output) ->
+    return unless input.hasData 'in'
+    return unless input.hasStream 'property'
+    data = input.getData 'in'
+    propData = input.getStream 'property'
+      .filter (ip) -> ip.type is 'data'
+      .map (ip) -> ip.data
+
     # Clone the object so that the original isn't changed
-    object = _.clone(object)
+    object = clone data
```

Comment:
Yes, probably. Can create unexpected side effects though as people are not used to controlling that behavior.